### ISSUE TITLE
How to get the fusion site in the isoform fasta sequence?

### ISSUE THREAD

**Issue by pailloufat-stack** [ConesaLab/SQANTI3#67]
Hi  ;

I'm looking for fusion genes. SQANTI3 ran well, but I would like now, to get the position of the fusion site in my fasta sequences.

For example, for the isoform PB.X.X , the fusion site is GTAG . If I awk/grep to look at this pattern in my fasta, it is found multiple times because of its short length.

I probably didn't find that information in the output files , or maybe would you have a solution to get it?

Best

**Comment by pailloufat-stack** [comment]
Any suggestion?

**Comment by Magdoll** [comment]
Hi @pailloufat-stack , I have an extra [post-processing script after running fusion w SQANTI3](https://github.com/Magdoll/cDNA_Cupcake/wiki/Best-practice-for-fusion-transcript-finding#collate-fusion-information-and-filtering-candidates) that should give you the fusion breakpoint. Is this what you're looking for?

**Comment by pailloufat-stack** [comment]
Hi @Magdoll ,

thanks for you reply. It is something like that. I guess, it is the information missing in the `*_junctions.txt` file, in the 7th column `transcript_coord` , after running the `sqanti3_qc.py` script . I would like to get the exact position of the fusion site in the transcript (for example, the fusion site is at the 437th nt of my transcript).  

I also noticed something. I don't get the same number of fusion transcripts after running `sqanti3_qc.py` and `fusion_finder.py` . I know, there is the collapsing step to run  `sqanti3_qc.py`  , but I got 47 fusion transcripts with that script and only one when I run `fusion_finder.py` . 

Which script would you better suggest me to find my fusion genes, and how could you explain a such difference?

Bests

**Comment by pailloufat-stack** [comment]
Hi @Magdoll , 

I'm coming back to you. I went further in order to spot those fusion sites in my fusion transcripts. 

1- I checked the `*junctions.txt` file to get the coordinates in 5' et 3' of the acceptor and donor sites. I put that in a bed file and I extracted +/- 15 nucleotides downstream and upstream to sport the motif GTAG for example in my transcript sequences. Impossible to spot them.

2- I aligned the transcripts on the reference genome. I looked at the CIGAR in my .sam files. I checked for them where there is only one "M" = mismatch ; so potentially the junction between my two fusion genes. I picked up the sequences downstream and upstream of that "M" , in fasta. I BLAST those two sequences on my reference and I didn't spot any "gap"/"junction" because they BLAST consecutively. 

3- I also BLAST the transcripts on my reference. For example, with the help of the GTF file , I can say the transcript PB.27.1 comes from two fusion genes : AT1G02050 (2exons) et AT1G02060 (1exon) . I looked at the concordance between the coordinated of my BLAST alignemnt and the coordinates of my genes in the GTF. The transcript BLAST well in its full length. But It is not coherent : for example, PB.27.1 BLAST before the first 5' exon of AT1G02050 (359,089 pb rather than 359,117 pb in the GTF)  and after the coordinate of the second exon in 3' (360,644 rather than 360,518 in the GTF). Also before in 5' with the mono-exon of AT1G02060. 
So, I supposed it was probably matching on the UTR regions. I downloaded the files to check the coherence and no! It is not matching on those regions.

So I don't understand.. Also, I tried the `fusion_finder.py` script, and the number of fusion genes found in drastically lower compared to the SQANTI3 pipeline (my previous message).

Any help?

Bests
